docs/source/usage.rst

```diff
@@ -725,3 +725,80 @@ Here is an example doing translation using a model and a tokenizer. The process
     outputs = model.generate(inputs, max_length=40, num_beams=4, early_stopping=True)
 
     print(outputs)
+Text Generation
```

Comment:
There is already a Causal Language Modeling section, which is basically text generation? It's currently under "Language Modeling" alongside Masked Language Modeling.

I think Text Generation is clearer than Causal Language Modeling, especially to newcomers. Maybe we could breakup Language Modeling so that it is only Masked Language Modeling, and put Text Generation below so that they are both on the highest level of the toc-tree?

Reply:
True, I didn't see that Causal Language Modeling also uses the `generate` method. 

Or should we maybe have three sections: 1) Masked Language Modeling - as it is now, 2) Causal Language Modeling - only show how the last token can be generated from the output distribution, 3) Text Generation - say that this is done automatically in `generate()` and show pipeline ...

Reply:
Sure that sounds good to me!

Reply:
Ok, I now added masked lm, causal lm and text generation all under language modeling. I think it's quite nice now. Can you take a look @LysandreJik ? :-)